torchvision/prototype/transforms/_meta_conversion.py

```diff
@@ -0,0 +1,50 @@
+from typing import Union, Any, Dict
+
+import torch
+import torchvision.prototype.transforms.kernels as K
+from torchvision.prototype import features
+from torchvision.prototype.transforms import Transform
+from torchvision.transforms.functional import convert_image_dtype
+
+
+class ConvertBoundingBoxFormat(Transform):
+    def __init__(self, format: Union[str, features.BoundingBoxFormat]) -> None:
+        super().__init__()
+        if isinstance(format, str):
+            format = features.BoundingBoxFormat[format]
+        self.format = format
+
+    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not isinstance(input, features.BoundingBox):
+            return input
+
+        output = K.convert_bounding_box_format(input, old_format=input.format, new_format=self.format)
+        return features.BoundingBox.new_like(input, output, format=self.format)
+
+
+class ConvertImageDtype(Transform):
+    def __init__(self, dtype: torch.dtype = torch.float32) -> None:
+        super().__init__()
+        self.dtype = dtype
+
+    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not isinstance(input, features.Image):
+            return input
+
+        output = convert_image_dtype(input, dtype=self.dtype)
+        return features.Image.new_like(input, output, dtype=self.dtype)
+
+
+class ConvertColorSpace(Transform):
+    def __init__(self, color_space: Union[str, features.ColorSpace]) -> None:
+        super().__init__()
+        if isinstance(color_space, str):
+            color_space = features.ColorSpace[color_space]
+        self.color_space = color_space
+
+    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not isinstance(input, features.Image):
+            return input
+
+        output = K.convert_color_space(input, old_color_space=input.color_space, new_color_space=self.color_space)
+        return features.Image.new_like(input, output, color_space=self.color_space)
```

Comment:
This should have been part of the high-level kernel I think. Most of them are supposed to return the right type with the right meta-data. Could you check that this is the case with all high-level kernels (convert_image_dtype, convert_bounding_box_format, etc)? Again I would recommend bringing this on a separate PR.

Reply:
Not sure what you are saying here:

- "Why do we have a `ConvertColorSpace` transform?": Because it might be useful to have a transform to convert to a specific color space even if all low-level kernels know how to handle the color space.
- "Why is there no high-level kernel for `convert_color_space`": I intentionally didn't implement any high-level kernels, i.e. dispatchers, for the type and meta conversions, since they by definition apply to only a single feature type. Thus, there wrapping them in a dispatcher is fruitless.

Reply:
As discuss offline, there PIL images can also be grayscaled so you have not a single feature type. But even in the case where there is only 1, the dispatcher will handle for you the type convention (`new_like`) and also the overall solution can benefit from the simplicity of always having a high-level kernel for all low-level kernels.

Reply:
We don't have any of the benefits of the automatic dispatch here, since all types need manual handling

```py
@dispatch(
    {
        torch.Tensor: None,
        PIL.Image.Image: None,
        features.Image: None,
    }
)
def convert_color_space(input: T, *args: Any, **kwargs: Any) -> T:
    color_space = kwargs["color_space"]
    if type(input) is torch.Tensor or isinstance(input, PIL.Image.Image):
        if color_space != "grayscale":
            raise ValueError("For vanilla tensors and PIL images only RGB to grayscale is supported")
        return _F.rgb_to_grayscale(input)
    elif isinstance(input, features.Image):
        output = K.convert_color_space(input, old_color_space=input.color_space, new_color_space=color_space)
        return features.Image.new_like(input, output, color_space=color_space)

    raise RuntimeError
```

Reply:
Where do you plan to put the logger if you don't have a high-level kernel? Note that the proposed solution should handle all of the needs outlined. Logging is currently one of them and maintaining a parity between low-level and high-level kernels can allow you to deal with this. If you have alternatives, I'm happy to discuss as long as we don't have to introduce a brand new mechanism.